templates/components/agents/typescript/form_filling/app/api/chat/workflow/factory.ts

```diff
@@ -0,0 +1,79 @@
+import { Message } from "ai";
+import { ChatMessage, ToolCallLLM } from "llamaindex";
+import { getAnnotations } from "../llamaindex/streaming/annotations";
+import { FormFillingWorkflow } from "./formFilling";
+import { getAvailableTools } from "./tools";
+
+const TIMEOUT = 360 * 1000;
+
+const prepareChatHistory = (chatHistory: Message[]): ChatMessage[] => {
```

Comment:
this is the same as in financial_report? should then be reused.
suggestion:
1. move to annotations.ts 
2. split in two functions: `convertToChatHistory` and `retrieveAgentHistory` (see note below)
3. call `convertToChatHistory` from route.ts (I added a note there)